libraries/microsoft-agents-hosting-core/microsoft_agents/hosting/core/app/oauth/_handlers/agentic_user_authorization.py

```diff
@@ -0,0 +1,148 @@
+
+logger = logging.getLogger(__name__)
+
+
+class AgenticUserAuthorization(_AuthorizationHandler):
+    """Class responsible for managing agentic authorization"""
+
+    def __init__(
+        self,
+        storage: Storage,
+        connection_manager: Connections,
+        auth_handler: Optional[AuthHandler] = None,
+        *,
+        auth_handler_id: Optional[str] = None,
+        auth_handler_settings: Optional[dict] = None,
+        **kwargs,
+    ) -> None:
+        """
+        Creates a new instance of Authorization.
+
+        :param storage: The storage system to use for state management.
+        :type storage: Storage
+        :param connection_manager: The connection manager for OAuth providers.
+        :type connection_manager: Connections
+        :param auth_handlers: Configuration for OAuth providers.
+        :type auth_handlers: dict[str, AuthHandler], optional
+        :raises ValueError: When storage is None or no auth handlers provided.
+        """
+        super().__init__(
+            storage,
+            connection_manager,
+            auth_handler,
+            auth_handler_id=auth_handler_id,
+            auth_handler_settings=auth_handler_settings,
+            **kwargs,
+        )
+        self._alt_blueprint_name = auth_handler._alt_blueprint_name if auth_handler else None
+        
+
+    async def get_agentic_instance_token(self, context: TurnContext) -> TokenResponse:
+        """Gets the agentic instance token for the current agent instance.
+
+        :param context: The context object for the current turn.
+        :type context: TurnContext
+        :return: The agentic instance token, or None if not an agentic request.
+        :rtype: Optional[str]
+        """
+
+        if not context.activity.is_agentic_request():
+            return TokenResponse()
+
+        assert context.identity
+        connection = self._connection_manager.get_token_provider(
+            context.identity, "agentic"
+        )
+        agentic_instance_id = context.activity.get_agentic_instance_id()
+        assert agentic_instance_id
+        instance_token, _ = await connection.get_agentic_instance_token(
+            agentic_instance_id
+        )
+        return TokenResponse(token=instance_token) if instance_token else TokenResponse()
+    
+
+
+    async def get_agentic_user_token(
+        self, context: TurnContext, scopes: list[str]
+    ) -> TokenResponse:
+        """Gets the agentic user token for the current agent instance and user.
+
+        :param context: The context object for the current turn.
+        :type context: TurnContext
+        :param scopes: The scopes to request for the token.
+        :type scopes: list[str]
+        :return: The agentic user token, or None if not an agentic request or no user.
+        :rtype: Optional[str]
+        """
+
+        if not context.activity.is_agentic_request() or not context.activity.get_agentic_user():
+            return TokenResponse()
```

Comment:
Consider throwing here. If this is called and these are not true, then there's an error somewhere above.